### PR TITLE
#225 - case studies

### LINKED ISSUE #225: [Dashboard] Case studies
We should implement CRUD for case studies on our website.

Case study should have fields like that:
- photo,
- project_name,
- company,
- text/description,
- template,
- slug,
- active,
- created_at/published_at.

Case study should be created for two languages - `pl` and `en`.

That case studies should be listed on that [page](https://xd.adobe.com/view/26f66876-cddd-469c-8b1f-b991dec40e55-73a5/screen/149a56ab-f403-4035-a608-4c0cbd75fcb7/). Details of specific case study should be created as a static template - static blade view. Created template should be assigned to the specific case study record.

### PR DESCRIPTION
**Pull request by kamilpiech97**
Should close #225.
Added case study model and CRUD views in admin panel. Some of case study fields are translatable via [laravel-translatable](https://spatie.be/docs/laravel-translatable/v6/introduction).

Preview:
image

### CODE REVIEW

File: app/Filament/Resources/CaseStudyResource.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                ])->from("lg"),
+            ])->columns(1);
+    }
+
+    public static function table(Table $table): Table
+    {
+        return $table
+            ->columns([
+                Tables\Columns\TextColumn::make("name")
+                    ->label("Nazwa relizacji")
```
Comment (EwelinaSkrzypacz) [inline app/Filament/Resources/CaseStudyResource.php:78]:
```suggestion
                    ->label("Nazwa realizacji")
```

**Review by EwelinaSkrzypacz — COMMENTED** [review]
When I create a case study without template I see error

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. When case study has no template, on index we should add dash, becaouse now it looks empty


image

2. When I edit case study - for example title field I have to refresh index to see changes - is it okay for us?

Flow:

You have a list of case studies with the title "best title ever" -> I edit this case study and give the title "Case study 1" ->click "Save" -> then "Cancel" to go back to the case study list page -> I still see "best title ever" -> when I refresh the page, only then do I see "Case study 1"

**Comment by kamilpiech97** [comment]
> 1. When case study has no template, on index we should add dash, becaouse now it looks empty
> 
> image
> 
> 2. When I edit case study - for example title field I have to refresh index to see changes - is it okay for us?
> 
> Flow:
> 
> You have a list of case studies with the title "best title ever" -> I edit this case study and give the title "Case study 1" ->click "Save" -> then "Cancel" to go back to the case study list page -> I still see "best title ever" -> when I refresh the page, only then do I see "Case study 1"

1, 2 fixed

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Review by KarolZygadlo — APPROVED** [review]

**Pull request merged**